gpkit/constraints/signomial_program.py

```diff
@@ -121,13 +123,28 @@ def localsolve(self, solver=None, verbosity=1, x0=None, rel_tol=1e-4,
         if verbosity > 0:
             print("Solving took %i GP solves" % len(self.gps)
                   + " and %.3g seconds." % (time() - starttime))
-        self.process_result(result)
         self.result = result  # NOTE: SIDE EFFECTS
         return result
 
+    def _default_x0(self, x0_in):
+        """Creates a default x0 dictionary.
+
+        Order of precedence for x0 default:
+            - substitution value
+            - x0 value
+            - sp_init value
+        """
+        x0 = KeyDict({vk: vk.sp_init for vk in self.varkeys if vk.sp_init})
+        x0.varkeys = self.varkeys  # check that keys are actually in the SP
+        if x0_in is not None:
+            x0.update(x0_in)
+        x0.update(self.substitutions)
+        return x0
+
     def gp(self, x0=None, verbosity=1):
         """Get a GP approximation of this SP at x0"""
+        x0 = self._default_x0(x0)
```

Comment:
maybe only do this if `x0 is None`? I believe our default in nomials is to not support a partially-filled x0; we could adopt that here too to avoid repeatedly creating the default each iteration.